common/utils_unittest.cpp

```diff
@@ -127,17 +127,6 @@ UTILS_TEST(guessResolutionOverflow) {
             << std::numeric_limits<long>::max();
 
     EXPECT_FALSE(guessResolution(sstream.str().c_str(), w, h));
-    EXPECT_EQ(w, 0);
-    EXPECT_EQ(h, 0);
-
-    sstream.str("");
-    sstream << long(std::numeric_limits<int>::max()) * 2 + 3
-            << "x"
-            << long(std::numeric_limits<int>::max()) * 3 + 2;
```

Comment:
This case is meant to test when an int overflow wraps back into a positive value.  On 64 bit system, `long(std::numeric_limits<int>::max()) * 2 + 3` == `4294967297` and `sscanf` in `guessResolution` would result in `1`.

Similarly, for `long(std::numeric_limits<int>::max()) * 3 + 2` == `6442450943` and `sscanf` would result in `2147483647`

Thus, consider the possibility that filename == `4294967297x4294967297`... `guessResolution` would return `true` and set the output vars `w=1` and `h=1`.  Such filename causes yamiencode to core dump.

Reply:
Hmm, I checked scanf return 1 for overflowed value. 
It's hard to handle this problem if scanf can't handle it well. 
Since long still have a problem on this. 
Any suggestion?

Reply:
`std::istringstream` might be a good solution:
```
std::istringstream iss("6442450943");
int value(0);
iss >> value;
assert !iss.fail()
```

Reply:
`std::stoi` in C++11 would be faster than `std::istringstream`

Reply:
Here's a good overview of the pros and cons of different techniques: http://www.kumobius.com/2013/08/c-string-to-int/

Reply:
For a simple/quick fix, we could just refactor like so:
```
diff --git a/common/utils.cpp b/common/utils.cpp
index 4aeb2da2989f..55c77314e678 100644
--- a/common/utils.cpp
+++ b/common/utils.cpp
@@ -74,8 +74,8 @@ bool guessResolution(const char* filename, int& w, int& h)
     int state = STATE_START;
     std::string strFilename(filename);
     const char* tokStart = NULL;
-    long width = 0;
-    long hight = 0;
+    int width = 0;
+    int hight = 0;
 
     w = h = 0;
     uint32_t i = 0;
@@ -94,7 +94,10 @@ bool guessResolution(const char* filename, int& w, int& h)
                 if (strFilename[i] == 'x' || strFilename[i] == 'X') {
                     state = STATE_X;
                     strFilename[i] = ' ';
-                    width = std::strtol(tokStart, NULL, 10);
+                    std::istringstream iss(tokStart);
+                    iss >> width;
+                    if (iss.fail())
+                      return false;
                 }
                 else if (!isdigit(strFilename[i])) {
                     state = STATE_START;
@@ -116,7 +119,10 @@ bool guessResolution(const char* filename, int& w, int& h)
                 if (!isdigit(strFilename[i])) {
                     state = STATE_END;
                     strFilename[i] = ' ';
-                    hight = std::strtol(tokStart, NULL, 10);
+                    std::istringstream iss(tokStart);
+                    iss >> hight;
+                    if (iss.fail())
+                      return false;
                 }
                 break;
             }
@@ -126,19 +132,16 @@ bool guessResolution(const char* filename, int& w, int& h)
     }
 
     //conner case
-    if (strFilename.length() == i && state == STATE_HEIGHT)
-        hight = std::strtol(tokStart, NULL, 10);
-
-    long maxInt = std::numeric_limits<int>::max();
-    if (width > maxInt || hight > maxInt) {
-        w = 0;
-        h = 0;
-    }
-    else {
-        w = width;
-        h = hight;
+    if (strFilename.length() == i && state == STATE_HEIGHT) {
+        std::istringstream iss(tokStart);
+        iss >> hight;
+        if (iss.fail())
+          return false;
     }
 
+    w = width;
+    h = hight;
+
     return w && h;
 }
```

...but, maybe it's better to refactor entire function, instead, for better efficiency?  Your choice.

Reply:
thanks, @uartie , it's really a good lesson. get so much information from your link. efficiency is not a major problem for this function. it only uses one/two time for one round.

Reply:
changed. please help review
thanks

Reply:
Since you fixed the overflow issue, please don't remove these checks in this unittest.  We still want to verify that passing > MAX_INT value returns false and does not regress in the future.  To fix 32-bit system where MAX_INT == MAX_LONG, we should use `long long` here instead (or any type that can hold > MAX_INT value).  Removing expectation that w,h == 0 is OK.